### PR TITLE
Adding a translator for idref.fr

### PR DESCRIPTION
**Pull request by symac**
Hello,
[idref.fr](http://www.idref.fr) is the website allowing to get information about authorities (identifiers) available in various French catalogs (mainly [SUDOC](http://www.sudoc.abes.fr/) and [HAL](https://hal.archives-ouvertes.fr/) for the moment). These linked catalogs already have existing translators so this translator is mainly calling the correct one based on the list of items existing on a page. 

Currently the translator is working as expected ([https://www.idref.fr/199676100](https://www.idref.fr/199676100) for example), but I have not been able to run the populated test, I imagine that this is related to the way I call selectItems but I have not been able to understand what the issue is), I hope the Zotero team will be able to fix this slight issue before merging.

Happy to update my code if any advice is given regarding things that would not be done according to Zotero's expectations.

### CODE REVIEW

File: Idref.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
+	"translatorID": "271ee1a5-da86-465b-b3a5-eafe7bd3c156",
+	"label": "Idref",
+	"creator": "Sylvain Machefert",
+	"target": "^https?://www\\.idref\\.fr/.*",
```
Comment (adam3smith) [inline Idref.js]: no need for the `.*` -- the target needs to match against the URL, not the other way around.

File: Idref.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "g",
```
Comment (adam3smith) [inline Idref.js]: `gcsibv`

File: Idref.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	"lastUpdated": "2018-02-03 09:39:41"
+}
+
```
Comment (adam3smith) [inline Idref.js]: Add licenese

File: Idref.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+function doWeb(doc, url)
+{
+	if (detectWeb(doc, url) == "multiple") {
+		var resultsTitle = ZU.xpath(doc, '//div[@id="perenne-references-docs"]/span[contains(@class, "detail_value")]');
+		var resultsHref = ZU.xpath(doc, '//div[@id="perenne-references-docs"]/span[contains(@class, "detail_label")]/a/@href');
+		
+		items = {};
+		for (var i in resultsTitle) {
```
Comment (adam3smith) [inline Idref.js]: iterate through arrays using either `for (let i=0; i<resultsTitle.lenght; i++)` or `for (result of resultsTitle)`

File: Idref.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+		});
+
+	}	
+}
+
+function scrape(doc, url) {
+	Z.debug("scraping : " + url);
+	var translator = Zotero.loadTranslator('web');	
+	
+	if (url.indexOf("archives-ouvertes")!=-1){
```
Comment (adam3smith) [inline Idref.js]: we've switched to using `includes` throughout for literal matches for readability (about same efficiency), so `if (url.includes("archives-ouvertes"))` etc.

File: Idref.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	if (detectWeb(doc, url) == "multiple") {
+		var resultsTitle = ZU.xpath(doc, '//div[@id="perenne-references-docs"]/span[contains(@class, "detail_value")]');
+		var resultsHref = ZU.xpath(doc, '//div[@id="perenne-references-docs"]/span[contains(@class, "detail_label")]/a/@href');
+		
+		items = {};
+		for (var i in resultsTitle) {
+			href = resultsHref[i].textContent;
+			// We need to replace the http://www.sudoc.fr/XXXXXX links are they are redirects and aren't handled correctly from subtranslator
+			href = href.replace(/http:\/\/www\.sudoc\.fr\/(.*)$/, "http://www.sudoc.abes.fr/xslt/DB=2.1//SRCH?IKT=12&TRM=$1");
+			items[href] = resultsTitle[i].textContent;
```
Comment (adam3smith) [inline Idref.js]: Why not filter out the URLs that you know we won't be able to import here? Better to not offer those results than to fail on them.

**Comment by symac** [comment]
Thanks @adam3smith . I have updated the file following your comments, should be better now. Tell me if you see anything else that could be improved.

File: Idref.js [hunk 7]
```diff
@@ -2,62 +2,85 @@
...
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2018-02-12 13:03:06"
 }
 
+/*
+   IdRef Translator
+   Copyright (C) 2018 Sylvain Machefert - web@geobib.fr
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
```
Comment (zuphilip) [inline Idref.js]: All Zotero translators should be licensed under AGPL (Affero), see https://github.com/zuphilip/translators/wiki/Common-code-blocks-for-translators for the relevant block.

Comment (symac) [reply Idref.js]: now fixed. I had copied another translator from the repo without having in mind this information regarding the license to be used. AGPL is also fine for me, thanks for the pointer.

**Comment by symac** [comment]
@adam3smith @zuphilip what is the next step regarding this translator? I believe I have fixed the requests both of you did previously, or do I just need to wait (which is fine, no hurry, just want to be sure I am not missing something that I could do).

**Comment by adam3smith** [comment]
Thanks!

**Comment by symac** [comment]
@adam3smith can you confirm that it is working as expecting for you? I have been told there were some issues and when testing, the translator is not showing in Firefox. If I try to visit [this page](https://www.idref.fr/199676100) in FF, nothing happens, if I visit the same page with Scaffold inside Zotero it is ok. 

I have just resinstalled everything (Zotero, Scaffold, Firefox connector). Any idea ? don't know if you have access to debug reports but in case, I have just created one under reference D1823926315.

**Comment by adam3smith** [comment]
Yes, there's something weird going on with detect. Reloading the page sometimes works for me but not always.

**Pull request merged**